backend/layers/api/portal_api.py

```diff
@@ -423,11 +437,13 @@ def publish_post(collection_id: str, body: object, token_info: dict):
     """
 
     version = get_business_logic().get_collection_version(CollectionVersionId(collection_id))
+    if version is None:
```

Comment:
non-blocking suggestion: I do think if we're going to be doing this double lookup so often, we should probably capture it in a double_lookup function similar to the curation API

Reply:
It was worth doing, I pushed it. Thanks!